### ISSUE TITLE
toset multiple external subnet unordered NIC assignment

### ISSUE THREAD

**Issue by JeffGiroux** [F5Networks/terraform-aws-bigip-module#26]
The AWS locals.tf file contains some external subnet ID mappings and result in the ext_interfaces. There is a loop in the dynamic "network_interface" block to attach the NICs to the VM. However, the 'toset' function is used and this has a random order. It is not consistent. Therefore, when you assign more than 1 external subnet in the field...the order of external NIC attachment to the VM is inconsistent.

## Expected Behavior
NIC assignment is in order of code

## Current behavior
Multiple external subnets...hence external NICs assigned will create an out of order assignment of NICs to the VM. Mappings are off and self IPs and MACs will not line up as expected.

sample code in question from main.tf
```
  # set the public interface only if an interface is defined
  dynamic "network_interface" {
    for_each = length(local.ext_interfaces) > 0 ? toset(local.ext_interfaces) : toset([])
    content {
      network_interface_id = network_interface.value
      device_index         = 1 + index(tolist(toset(local.ext_interfaces)), network_interface.value)
    }
  }
```

https://www.terraform.io/language/functions/toset
"Set collections are unordered and cannot contain duplicate values, so the ordering of the argument elements is lost and any duplicate values are coalesced:"

## Example code

Here is code that uses multiple external subnets and security groups. The 'toset' function causes a random order of assignment.
```
module "bigip" {
  source                     = "F5Networks/bigip-module/aws"
  external_subnet_ids        = [local.ext_subnets.external, local.ext_subnets.dmz1, local.ext_subnets.dmz2, local.ext_subnets.dmz3, local.ext_subnets.dmz4]
  external_securitygroup_ids = [local.ext_security_groups.external, local.ext_security_groups.dmz1, local.ext_security_groups.dmz2, local.ext_security_groups.dmz3, local.ext_security_groups.dmz4]
}
```

I would expect my NICs to be in the following order
0. external
1. dmz1
2. dmz2
3. dmz3
4. dmz4

However, with 'toset'...my current BIG-IP deployed has the following order

0. dmz3
1. dmz4
2. external
3. dmz2
4. dmz1

**Comment by JeffGiroux** [comment]
This affects the private network_interfaces too.

```
  dynamic "network_interface" {
    for_each = length(aws_network_interface.private) > 0 ? toset([aws_network_interface.private[0].id]) : toset([])


  dynamic "network_interface" {
    for_each = length(aws_network_interface.private1) > 0 ? toset([aws_network_interface.private1[0].id]) : toset([])
```

And mgmt
```
  dynamic "network_interface" {
    for_each = length(compact(local.mgmt_public_private_ip_primary)) > 0 ? toset([aws_network_interface.mgmt[0].id]) : toset([aws_network_interface.mgmt1[0].id])

```

**Comment by JeffGiroux** [comment]
## Workaround
Here is the fix. You need to remove 'toset' from all network_interface block logic. You need to remove it in both lines...the for_each line AND the device_index. This fix needs to be applied respectively to all network interfaces.

```
  dynamic "network_interface" {
    for_each = length(local.ext_interfaces) > 0 ? local.ext_interfaces : null
    content {
      network_interface_id = network_interface.value
      device_index         = 1 + index(tolist(local.ext_interfaces), network_interface.value)
    }
  }
```

**Comment by RavinderReddyF5** [comment]
@JeffGiroux fixed in v1.1.7, please verify
